h2o-core/src/main/java/water/JettyHTTPD.java

```diff
@@ -305,7 +305,13 @@ public void registerHandlers(HandlerWrapper handlerWrapper) {
     ServletContextHandler context = new ServletContextHandler(
         ServletContextHandler.SECURITY | ServletContextHandler.SESSIONS
     );
-    context.setContextPath("/");
+
+    if(null != H2O.ARGS.context_path && !H2O.ARGS.context_path.isEmpty()) {
+      context.setContextPath(H2O.ARGS.context_path);
```

Comment:
If you specify path `java -jar h2o.jar -context_path xxx` then h2o is not going to be available at `localhost:54321/xxx`. Furthermore, the code should remove all `/` at the end of the path since they are not allowed